csharp/src/Ice/Internal/RouterInfo.cs

```diff
@@ -84,21 +84,25 @@ public void getClientEndpoints(GetClientEndpointsCallback callback)
             callback.setEndpoints(clientEndpoints);
             return;
         }
+        _ = performGetClientProxyAsync(_router);
 
-        _router.getClientProxyAsync().ContinueWith(
-            (t) =>
+        async Task performGetClientProxyAsync(RouterPrx router)
```

Comment:
Task.Run seems simpler here too.

Reply:
They are more or less the same, I prefer to use local functions everywhere for consistency.